docker/gcp_pudl_etl.sh

```diff
@@ -100,60 +95,95 @@ function notify_slack() {
-    echo "BOGUS: Updating nightly branch to point at $NIGHTLY_TAG."
-    git fetch origin nightly:nightly
-    git checkout nightly
-    git merge --ff-only "$NIGHTLY_TAG"
-    ETL_SUCCESS=${PIPESTATUS[0]}
-    git push -u origin
+    git config --unset http.https://github.com/.extraheader && \
+    git config user.email "pudl@catalyst.coop" && \
+    git config user.name "pudlbot" && \
+    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git" && \
+    echo "Updating nightly branch to point at $NIGHTLY_TAG." && \
+    git tag -d "$NIGHTLY_TAG" && \
+    git fetch --tags origin "$NIGHTLY_TAG" && \
+    git checkout "$NIGHTLY_TAG" && \
+    git tag && \
+    git fetch origin nightly:nightly && \
+    git checkout nightly && \
+    git merge --ff-only "$NIGHTLY_TAG^0" && \
+    git push -u origin nightly
+}
+
+function clean_up_outputs_for_distribution() {
+    # Compress the SQLite DBs for easier distribution
+    gzip --verbose "$PUDL_OUTPUT"/*.sqlite && \
+    # Remove redundant multi-file EPA CEMS outputs prior to distribution
+    rm -rf "$PUDL_OUTPUT/core_epacems__hourly_emissions/" && \
+    rm -f "$PUDL_OUTPUT/metadata.yml"
 }
 
-# # Run ETL. Copy outputs to GCS and shutdown VM if ETL succeeds or fails
+########################################################################################
+# MAIN SCRIPT
+########################################################################################
+# Initialize our success variables so they all definitely have a value to check
+ETL_SUCCESS=0
+SAVE_OUTPUTS_SUCCESS=0
+UPDATE_NIGHTLY_SUCCESS=0
+DATASETTE_SUCCESS=0
+CLEAN_UP_OUTPUTS_SUCCESS=0
+DISTRIBUTION_BUCKET_SUCCESS=0
+ZENODO_SUCCESS=0
+
+# Set PUDL_GCS_OUTPUT *only* if it is currently unset
+: "${PUDL_GCS_OUTPUT:=gs://nightly-build-outputs.catalyst.coop/$BUILD_ID}"
+
+# Run ETL. Copy outputs to GCS and shutdown VM if ETL succeeds or fails
 # 2>&1 redirects stderr to stdout.
 run_pudl_etl 2>&1 | tee "$LOGFILE"
 ETL_SUCCESS=${PIPESTATUS[0]}
 
-copy_outputs_to_gcs
+save_outputs_to_gcs 2>&1 | tee -a "$LOGFILE"
+SAVE_OUTPUTS_SUCCESS=${PIPESTATUS[0]}
 
 # if pipeline is successful, distribute + publish datasette
 if [[ $ETL_SUCCESS == 0 ]]; then
-    # Deploy the updated data to datasette
-    if [ "$BUILD_REF" = "dev" ]; then
-        python ~/pudl/devtools/datasette/publish.py 2>&1 | tee -a "$LOGFILE"
-        ETL_SUCCESS=${PIPESTATUS[0]}
+    if [[ "$GITHUB_ACTION_TRIGGER" == "schedule" ]]; then
+        update_nightly_branch 2>&1 | tee -a "$LOGFILE"
+        UPDATE_NIGHTLY_SUCCESS=${PIPESTATUS[0]}
     fi
 
-    # Compress the SQLite DBs for easier distribution
-    # Remove redundant multi-file EPA CEMS outputs prior to distribution
-    gzip --verbose "$PUDL_OUTPUT"/*.sqlite && \
-    rm -rf "$PUDL_OUTPUT/core_epacems__hourly_emissions/" && \
-    rm -f "$PUDL_OUTPUT/metadata.yml"
-    ETL_SUCCESS=${PIPESTATUS[0]}
```

Comment:
Moved this into its own function, and put it inside the conditional below which only happens if we're doing distribution.